frame/balances/src/benchmarking.rs

```diff
@@ -28,46 +28,39 @@ use sp_runtime::traits::Bounded;
 use crate::Module as Balances;
 
 const SEED: u32 = 0;
-const MAX_EXISTENTIAL_DEPOSIT: u32 = 1000;
-const MAX_USER_INDEX: u32 = 1000;
+const ED_MULTIPLIER: u32 = 10;
+
 
 benchmarks! {
-	_ {
-		let e in 2 .. MAX_EXISTENTIAL_DEPOSIT => ();
-		let u in 1 .. MAX_USER_INDEX => ();
-	}
+	_ { }
 
 	// Benchmark `transfer` extrinsic with the worst possible conditions:
 	// * Transfer will kill the sender account.
 	// * Transfer will create the recipient account.
 	transfer {
-		let u in ...;
-		let e in ...;
-
 		let existential_deposit = T::ExistentialDeposit::get();
-		let caller = account("caller", u, SEED);
+		let caller = account("caller", 0, SEED);
 
 		// Give some multiple of the existential deposit + creation fee + transfer fee
-		let balance = existential_deposit.saturating_mul(e.into());
+		let balance = existential_deposit.saturating_mul(ED_MULTIPLIER.into());
 		let _ = <Balances<T> as Currency<_>>::make_free_balance_be(&caller, balance);
 
 		// Transfer `e - 1` existential deposits + 1 unit, which guarantees to create one account, and reap this user.
-		let recipient: T::AccountId = account("recipient", u, SEED);
+		let recipient: T::AccountId = account("recipient", 0, SEED);
 		let recipient_lookup: <T::Lookup as StaticLookup>::Source = T::Lookup::unlookup(recipient.clone());
-		let transfer_amount = existential_deposit.saturating_mul((e - 1).into()) + 1.into();
-	}: _(RawOrigin::Signed(caller), recipient_lookup, transfer_amount)
+		let transfer_amount = existential_deposit.saturating_mul((ED_MULTIPLIER - 1).into()) + 1.into();
+	}: transfer(RawOrigin::Signed(caller.clone()), recipient_lookup, transfer_amount)
 	verify {
+		assert_eq!(Balances::<T>::free_balance(&caller), Zero::zero());
 		assert_eq!(Balances::<T>::free_balance(&recipient), transfer_amount);
 	}
 
 	// Benchmark `transfer` with the best possible condition:
 	// * Both accounts exist and will continue to exist.
+	#[extra]
 	transfer_best_case {
-		let u in ...;
-		let e in ...;
-
-		let caller = account("caller", u, SEED);
-		let recipient: T::AccountId = account("recipient", u, SEED);
+		let caller = account("caller", 0, SEED);
```

Comment:
This is the new `#[extra]` tag syntax introduced in this PR which allows the benchmarking pipeline to skip this benchmark by default when generating stuff.

Reply:
If your goal is to distinguish between best case and worst case, maybe name the flag accordingly and skip best case by default?
(I'm still unsure how the API should be designed as well.)

Reply:
The goal is to skip a benchmark that is not used in the final weights of the Balances pallet. In this case the benchmark for "transfer_best_case" was just used to inform us, but never used in the pallet because we always assume transfer is worst case.